driver-mapping/src/main/java/com/datastax/driver/mapping/QueryType.java

```diff
@@ -69,6 +70,11 @@ String makePreparedQueryString(TableMetadata table, EntityMapper<?> mapper, Mapp
                 Insert.Options usings = insert.using();
                 for (Mapper.Option opt : options) {
                     opt.checkValidFor(QueryType.SAVE, manager);
+
+                    if (shouldIfNotExists(opt)) {
```

Comment:
Maybe we should change the signature of `appendTo` to take `Insert/Delete` in parameter instead of `Insert.Options/Delete.Options`, so that you don't have to do a special case here and can apply `ifNotExists()` in `appendTo`.